custom/staking/src/lib.rs

```diff
@@ -1624,15 +1725,27 @@ impl<T: Trait> Module<T> {
         })
     }
 
-    /// Select a new validator set from the assembled stakers and their role preferences.
+    /// Select a new validator set from the assembled stakers and their role preferences. Validator
+    /// accounts are selected if they meet the following conditions:
+    ///  - active balance is at least `MinimumBondThreshold`
     ///
     /// Returns the new `SlotStake` value and a set of newly selected _stash_ IDs.
     ///
     /// Assumes storage is coherent with the declaration.
     fn select_validators() -> (BalanceOf<T>, Option<Vec<T::AccountId>>) {
         let mut all_nominators: Vec<(T::AccountId, Vec<T::AccountId>)> = Vec::new();
-        let all_validator_candidates_iter = <Validators<T>>::enumerate();
-        let all_validators = all_validator_candidates_iter
+        let all_validator_candidates = <Validators<T>>::enumerate()
+            .filter(|(who, _prefs)| {
+                if let Some(controller) = Self::bonded(&who) {
+                    if let Some(ledger) = Self::ledger(&controller) {
```

Comment:
Shouldn't the compliance be also checked here?

Reply:
We should. That's in the other currently open PR. It was part of older version of staking that didn't make into this module during v2 upgrade.